rust/nasl-interpreter/src/built_in_functions/frame_forgery.rs

```diff
@@ -0,0 +1,797 @@
+/// - frame: the frame to send, created with forge_frame
+/// - pcap_active: option to capture the answer, default is TRUE
+/// - pcap_filter: filter for the answer
+/// - pcap_timeout: time to wait for the answer in seconds, default 5
+fn nasl_send_frame(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let frame = match register.named("frame") {
+        Some(ContextType::Value(NaslValue::Data(x))) => x,
+        _ => {
+            return Err(FunctionError::new(
+                "send_frame",
+                ("Data", "Invalid data type").into(),
+            ))
+        }
+    };
+
+    let pcap_active = match register.named("pcap_active") {
+        Some(ContextType::Value(NaslValue::Boolean(x))) => x,
+        None => &true,
+        _ => {
+            return Err(FunctionError::new(
+                "send_frame",
+                ("Boolean", "Invalid pcap_active value").into(),
+            ))
+        }
+    };
+
+    let filter = match register.named("filter") {
+        Some(ContextType::Value(NaslValue::String(x))) => Some(x),
+        None => None,
+        _ => {
+            return Err(FunctionError::new(
+                "send_frame",
+                ("String", "Invalid pcap_filter value").into(),
+            ))
+        }
+    };
+
+    let timeout = match register.named("timeout") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as i32 * 1000i32, // to milliseconds
+        None => DEFAULT_TIMEOUT,
+        _ => {
+            return Err(FunctionError::new(
+                "send_frame",
+                ("Integer", "Invalid timeout value").into(),
+            ))
+        }
+    };
+
+    let target_ip = get_host_ip(register)?;
+
+    let local_ip = get_source_ip(target_ip, 50000u16)?;
+    let iface = get_interface_by_local_ip(local_ip)?;
+
+    // send the frame and get a response if pcap_active enabled
+    match send_frame(frame, &iface, pcap_active, filter, timeout)? {
+        Some(f) => Ok(NaslValue::Data(f.into())),
+        None => Ok(NaslValue::Null),
+    }
+}
+
+/// Print a datalink layer frame in its hexadecimal representation.
+/// The named argument frame is a string representing the datalink layer frame. A frame can be created with forge_frame(3).
+/// This function is meant to be used for debugging.
+fn nasl_dump_frame(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let frame: Frame = match register.named("frame") {
+        Some(ContextType::Value(NaslValue::Data(x))) => <&[u8]>::from(x).try_into()?,
+        _ => {
+            return Err(FunctionError::new(
+                "forge_frame",
+                ("Data", "Invalid data type").into(),
+            ))
+        }
+    };
+
+    // TODO: make NaslLogger to get a generic value, not only string
+    //let logger = DefaultLogger::new();
+    //logger.set_mode(Mode::Debug);
+    //logger.info(frame);
+    println!("Frame:\n{}", frame);
```

Comment:
I think it would be better to use:
```
register.logger().info(format!("Frame:\n{}", frame));
```
than using println and maybe remove the other comments.